src/Nethermind/Nethermind.Synchronization/SyncServer.cs

```diff
@@ -181,34 +181,41 @@ public void AddNewBlock(Block block, ISyncPeer nodeWhoSentTheBlock)
 
             bool isBlockBeforeTheSyncPivot = block.Number < _pivotNumber;
             bool isBlockOlderThanMaxReorgAllows = block.Number < (_blockTree.Head?.Number ?? 0) - Sync.MaxReorgLength;
-            bool isBlockTotalDifficultyLow = block.TotalDifficulty < _blockTree.BestSuggestedHeader.TotalDifficulty
-                                             && (_specProvider.TerminalTotalDifficulty == null || block.TotalDifficulty < _specProvider.TerminalTotalDifficulty); // terminal blocks with lower TTD might be useful for smooth merge transition
-            if (isBlockBeforeTheSyncPivot || isBlockTotalDifficultyLow || isBlockOlderThanMaxReorgAllows) return;
 
-            lock (_recentlySuggested)
+            if (isBlockBeforeTheSyncPivot || isBlockOlderThanMaxReorgAllows)
             {
-                if (_recentlySuggested.Get(block.Hash) != null) return;
-                _recentlySuggested.Set(block.Hash, _dummyValue);
+                return;
             }
 
-            if (_specProvider.TerminalTotalDifficulty != null && block.TotalDifficulty >= _specProvider.TerminalTotalDifficulty)
+            if (_recentlySuggested.Set(block.Hash))
```

Comment:
we removed lock(_recentlySuggested). Is that okay?

Reply:
and we changed a bit logic here

Reply:
yes, we don't use Get and Set separately. The result of Set should be enough for us and Set itself is thread-safe so we don't need to have an outside lock.